### PR TITLE
Allow configuration of spacer options

### PR DESCRIPTION
**Pull request by jhunschejones**
### Description:
I found myself wanting to use the built in spacers in Bulma to narrow down the page numbers displayed on a page in a way that would make intuitive sense to the user. In order to do that, I added the `maximum_links` config option. This allows the first and last page number to show, along with a max range of page numbers around the current page (as the user advances through the pages.) 

The code changes I made respect the defaults that were in this module before, only applying the change when the config option is set. The end result will look like this (screenshot of my app where I'm using this):
 
maximum_links_2_example

### Notes:
* I'm pretty sure the link to the example image should work once this is merged, but if you know more about github link markdown please let me know.
* If you have a moment to take a look through the algorithms I added to see if any improvement opportunities stick out to you, I'd be happy for the feedback!

### CODE REVIEW

File: lib/bulma_pagination/bulma_renderer.rb [hunk 1]
```diff
@@ -23,9 +21,34 @@ def container_attributes
...
-      link_options = @options[:link_options] || {}
+    def visible_page_numbers(pages)
+      if @options[:maximum_links] && pages.size > @options[:maximum_links]
+        link_offset = @options[:maximum_links] / 2
+        link_start = [(current_page.to_i - link_offset - 1), 0].max
+        link_end = current_page.to_i + link_offset - 1
+        pages = ([pages.first] + pages[link_start..link_end] + [pages.last]).uniq
+        pages = pagination_links_with_gaps(pages)
+      else
+        pages.map { |page| pagination_link(page) }
```
Comment (jhunschejones) [inline lib/bulma_pagination/bulma_renderer.rb:32]: Default behavior if new config option doesn't exist 👍

File: README.markdown [hunk 2]
```diff
@@ -61,7 +61,9 @@ The supported configuration options are as follows:
 * `link_options` a hash of attributes added to active pagination-link elements _(note, classes added here will be overwritten)_
 * `previous_label` text or html to replace the inner content of the link within the pagination-previous element
 * `next_label` text or html to replace the inner content of the link within the pagination-next element
+* `maximum_links` affect how spacers are applied when you have many pages. This allows you to show the first and last page, then the current page with a certain number of pages on either sides.
 
+![](https://github.com/phoffer/will_paginate-bulma/blob/master/example/maximum_links_2_example.png)
```
Comment (jhunschejones) [inline README.markdown:66]: This _should_ work once merged 🤔

**Comment by jhunschejones** [comment]
@phoffer, I hope all is well with you! I wanted to check in and see if there is anything else I can adjust on this PR for it to be ready to merge. Thanks for your time.

**Comment by phoffer** [comment]
> @phoffer, I hope all is well with you! I wanted to check in and see if there is anything else I can adjust on this PR for it to be ready to merge. Thanks for your time.

Hi! I'm sorry, I just forgot to merge and got distracted. Thank you for the update!

**Pull request merged**